### ISSUE TITLE
Cannot send email (smtp_domain stays at example.com)

### ISSUE THREAD

**Issue by flesser** [bitnami/bitnami-docker-discourse#31]
**Description**

In my `docker-compose.yml`, I have set `SMTP_HOST: smtp.MYDOMAIN.de`.
(I also changed the default Postgresql username and password)

After `docker-compose up`, Discourse runs, but cannot send any email.

`/opt/bitnami/discourse/conf/discourse.conf` inside the running container looks like this:
```yml
  db_name = discourse
  db_host = postgresql
  db_port = 5432
  db_password = PASSWORD
  db_username = discourse
  db_pool = 25
  redis_port = 6379
  redis_host = redis
  redis_password = 
  smtp_address = smtp.MYDOMAIN.de
    smtp_port = 587
    smtp_domain = example.com
    smtp_user_name = discourse@MYDOMAIN.de
    smtp_password =  PASSWORD
    smtp_enable_start_tls = true
    smtp_authentication = login
```
Two things I noticed:
* the indentation looks strange
*  `smtp_address` is set correctly to the domain I configured in docker-compose, but `smtp_domain` is still hardcoded to example.com

Is this intended? I would expect that this paramter is also configurable via the `docker-compose.yml`

**Comment by dbarranco** [comment]
Hi @flesser 

I just checked our daily tests of SMTP and they're working fine. Anyway, I added the env vars for the docker-compose to enable SMTP and this is how it looks like:

```
version: '2'
services:
  postgresql:
    image: 'bitnami/postgresql:latest'
    volumes:
      - 'postgresql_data:/bitnami'
  redis:
    image: 'bitnami/redis:latest'
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - 'redis_data:/bitnami'
  discourse:
    image: 'bitnami/discourse:1'
    labels:
      kompose.service.type: nodeport
    ports:
      - '80:3000'
    depends_on:
      - postgresql
      - redis
    volumes:
      - 'discourse_data:/bitnami'
    environment:
      - POSTGRESQL_HOST=postgresql
      - POSTGRESQL_ROOT_USER=postgres
      - POSTGRESQL_CLIENT_CREATE_DATABASE_NAME=bitnami_application
      - POSTGRESQL_CLIENT_CREATE_DATABASE_USERNAME=bn_discourse
      - POSTGRESQL_CLIENT_CREATE_DATABASE_PASSWORD=bitnami1
      - DISCOURSE_POSTGRESQL_NAME=bitnami_application
      - DISCOURSE_POSTGRESQL_USERNAME=bn_discourse
      - DISCOURSE_POSTGRESQL_PASSWORD=bitnami1
      - SMTP_HOST=smtp.gmail.com
      - SMTP_PORT=587
      - SMTP_USER=testsmtp@gmail.com
      - SMTP_PASSWORD=PASSWORD
      - SMTP_TLS=true
  sidekiq:
    image: 'bitnami/discourse:1'
    depends_on:
      - discourse
    volumes:
      - 'sidekiq_data:/bitnami'
    command: 'nami start --foreground discourse-sidekiq'
    environment:
      - DISCOURSE_POSTGRESQL_NAME=bitnami_application
      - DISCOURSE_POSTGRESQL_USERNAME=bn_discourse
      - DISCOURSE_POSTGRESQL_PASSWORD=bitnami1
      - DISCOURSE_HOST=discourse
      - DISCOURSE_PORT=3000
volumes:
  postgresql_data:
    driver: local
  redis_data:
    driver: local
  discourse_data:
    driver: local
  sidekiq_data:
    driver: local
```

Then I launched the container and in the settings of SMTP the email is correctly configured and a test email was successfully sent. 
Could you share here your configuration? Please, hide any password or sensitive data that is present in your docker-compose file.

Kind regards.

**Comment by macropin** [comment]
@dbarranco `smtp_domain = example.com` is hardcoded in `/root/.nami/components/com.bitnami.discourse/helpers.js`

**Comment by carrodher** [comment]
I got it thanks. We have included it in our backlog. As the impact of the issue is limited, there are no immediate plans to work on it. Nevertheless, it will be considered for future application versions.

We will update this thread as soon as we have more news.

**Comment by rene-gomez** [comment]
@macropin How can i modify this file?

best regards

**Comment by rene-gomez** [comment]
@carrodher Hi, so sorry but I think that issue is a very impact, because, never new user receive an confirmation email for signup, and the images does not work because redirect to example.com

**Comment by javsalgar** [comment]
@rene-gomez as mentioned here (https://github.com/bitnami/bitnami-docker-discourse/issues/41#issuecomment-385374568), did you try doing rake db:migrate?

**Comment by macropin** [comment]
@rene-gomez After the initial bootstrap you can edit the config file. 

As aside it seems the bitnami images do not reconcile configured state with actual state, hence the bootstrap only runs once. IMHO this is an anti-pattern, but in this case it works in our favour.

**Comment by Raphty** [comment]
hey, I had the same Problem.

for me the solution was to change the config file in the sidekiq as well.
/sidekiq/discourse-sidekiq/conf/discourse.conf


` hostname = ex.com
db_name = discourse
db_host = postgresql
db_port = p#
db_password = pw
db_username = UN
db_pool = 25
redis_port = p#
redis_host = redis
redis_password =
smtp_address = smtp.ex.com
smtp_port = p#
smtp_domain = DomainName
smtp_user_name = email@ex.com
smtp_password =  PW
smtp_enable_start_tls = true`

**Comment by mikaelhg** [comment]
I have a similar issue, where I first created the image without SMTP configuration, with volumes on the host drive, then added it, but no matter what environmental variables I give the image, the discourse.conf never gets an update.

Where can we find the `helpers.js` source code on GitHub? It seems to be the crux of the issue, but I just can't find it. Is it somehow autogenerated from a configuration file, or something?

**Comment by juan131** [comment]
Hi @mikaelhg 

As you mentioned the configuration is persisted but you can edit with any editor through the terminal. For instance, in order to edit it using vim, you can do:

```
$ docker-compose exec discourse install_packages vim
$ docker-compose exec discourse vim /bitnami/discourse/conf/discourse.conf
```

**Comment by juan131** [comment]
> Just to verify, do you mean, @juan131, that by design, Bitnami image configuration is run only once?

On this image, yes. That's something we're modifying on some Docker images, but this image hasn't been modified yet. 

Originally, we persisted data&configuration on every image. Then, we realised that it made sense to only persist data so the configuration is re-built every time the container is restarted.

Some other images, such as PostgreSQL, also support including your own configuration file. When it detects that your custom configuration file is mounted, it ignores the env. variables and use the one you mounted. See:

https://github.com/bitnami/bitnami-docker-postgresql#configuration-file
